interview-questions.md

```diff
@@ -5,55 +5,55 @@ First, without external resources. Challenge yourself to answer from memory.
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
-1. What is a block in Ruby?
+1. What is a block in Ruby? 
 
-  Your answer:
+  Your answer: A block in Ruby is a packet of information inside a component nesting in something
 
-  Researched answer:
+  Researched answer: The code that is passed to a method is called a block. Anonymous functions are also a block
 
 
 
 2. What is a gem?
 
-  Your answer:
+  Your answer:A jem is also a packet of information, but a series of specific packets that presume a function. 
 
-  Researched answer:
+  Researched answer: The standard package manager for distributing ruby programs and libraries is called Ruby Gems
 
 
 
 3. What is Ruby on Rails?
 
-  Your answer:
+  Your answer: An ecosystem that makes it easier to produce with a set of predefined funciton.
 
-  Researched answer:
+  Researched answer: Its a framework of library that simplifies web development 
 
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: a relational database is a database that is tied to another datatbase with a ID
 
-  Researched answer:
+  Researched answer: A database that is originated by relation to the other elements. Its important because it is the most used database structure in the use
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: primary keys are a automatically assigned variable 
```

Comment:
Primary keys aren't necessarily a variable. In Rails, the primary key does get automatically assigned every time we create a new instance in the db. It is important to ensure all db instances are unique.